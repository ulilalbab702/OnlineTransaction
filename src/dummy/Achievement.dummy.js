export const Achievement = [
    {
        "Code": "FM-JBYPM",
        "Name": "FMC Jembayan PAMA",
        "Percent": 50,
        "BackLogOpen": 1,
        "BackLogClose": 1,
        "Priority1": 1,
        "Priority2": 0,
        "Priority3": 1,
        "Priority4": 0
    }
]