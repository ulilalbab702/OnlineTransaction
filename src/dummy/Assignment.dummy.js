export const listOfMechanicDummy = [
    {
      "Username": "string",
      "FirstName": "string",
      "LastName": "string",
      "Email": "string",
      "MechanicUT": true,
      "Jobs": [
        {
          "JobType": "string",
          "Total": 0
        }
      ]
    }
  ]
export const jobSummaryDummy = {
  "Date": "2020-06-04T16:18:05.5472055",
  "CreatedBy": "AGUS LISANTO",
  "AssignedBy": "AGUS LISANTO",
  "MechanicLeader": {
      "UserName": "82107038@unitedtractors.com",
      "Name": "SUPRIYADI SUPRIYADI"
  },
  "Mechanics": [],
  "HandoverDate": null,
  "HandoverCreatedBy": null,
  "HandoverAssignedBy": null,
  "HandoverMechanicLeader": null,
  "HandoverMechanics": null,
  "Job": {
      "WoId": "8ff577d1-2911-4bbd-9d55-ffad4fe76fc5",
      "WoNumber": "71018650",
      "JobType": "INS",
      "UnitModel": "HD465-7",
      "SerialNumber": "15163",
      "UnitCode": "HT3022",
      "EquipmentNumber": "71506",
      "IsHandover": false,
      "Id": "a162bd06-6aca-499a-9f97-a5839b8a0612",
      "CreatedBy": "80107002@unitedtractors.com",
      "CreatedDate": "2020-06-04T16:18:05.5472055",
      "UpdatedBy": "80107002@unitedtractors.com",
      "UpdatedDate": "2020-06-04T16:18:05.5472055",
      "IsActive": true
  }
}