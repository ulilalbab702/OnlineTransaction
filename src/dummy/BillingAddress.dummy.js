export const dataBillingAddressDummy = [
    {
        "fullName": "Test Integrasi",
        "contactNumber": "088122345",
        "email": "testintegrasi@unitedtractors.com",
        "position": "string",
        "addressLabel": "Kantor PT Global Service Indonesia",
        "city": "Jakarta",
        "address": "Jl Raya kaliabang Tengah, Ruko San Jaya, Blok A5 No 9,\r\nSebalah Indomaret",
        "postalCode": "13920",
        "locationLatitude": "3.669276961596295",
        "locationLongitude": "102.4003031491451",
        "isPrimary": true
    }]