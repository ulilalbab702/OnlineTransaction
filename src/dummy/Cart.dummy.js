export const cartDummyList = {
    "cartProductResponse": {
        "data": [
            {
                "cartId": "d7e23521-0d23-4f37-aceb-77f8f75b3d16",
                "productId": "32bdd633-932f-465a-b93c-219a70971008",
                "name": "600-311-8291",
                "imageThumbnailUrl": "https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa",
                "qty": 3,
                "priceAfter": 300000,
                "priceBefore": 500000,
                "currency": "IDR",
                "cookiesData": ""
            },
            {
                "cartId": "10d112f9-0a0b-4c26-87b6-87ffb0374a77",
                "productId": "4211034d-8571-4ed6-84b2-8b380443bf7b",
                "name": "568002",
                "imageThumbnailUrl": "https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa",
                "qty": 2,
                "priceAfter": 800000,
                "priceBefore": 1000000,
                "currency": "IDR",
                "cookiesData": ""
            }
        ],
        "meta": {
            "totalItems": 2,
            "pageNumber": 1,
            "pageSize": 5,
            "totalPages": 1,
            "hasPreviousPage": false,
            "hasNextPage": false,
            "nextPageNumber": 1,
            "previousPageNumber": 1
        }
    },
    "subTotalPrice": 2500000
}
