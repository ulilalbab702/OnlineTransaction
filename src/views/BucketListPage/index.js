import BucketListPage from "./BucketListPage";
import { connect } from "react-redux";
import { LOGIN_TYPE } from "actions/actionTypes";
import { login } from "actions/user";
import { push } from "connected-react-router";
import { createLoadingSelector } from "utils/selector.helper";
import {
  listCart,
  searchByUserId,
  getListCartByCartId,
  putListCartByCartId,
  getPurchaseSummary,
  deleteAllCart,
  deleteCartDetail,
  getPurchaseCheckout,
  getCheckoutProduct,
  getAttributeName
} from "actions/cart";
import { getBranchList, getCheckStock } from "actions/listNews";

//Set selector loading
const loadingSelector = createLoadingSelector([LOGIN_TYPE]);
const mapStateToProps = (state) => ({
  ...state.user,
  isLoading: loadingSelector(state),
  cartList: state.cart.cart.data,
  cartListError: state.cart.cart.err,
  dataCart: state.cart.searchByUserId.dataCart,
  dataCartList: state.cart.listCart.dataCartList,
  purchaseSummary: state.cart.purchaseSummary.data,
  stockList: state.listNews.listNews.stockList,
  dataPut: state.cart.listCart.dataPutCartList,
  branchList: state.listNews.listNews.branchList,
  loading: state.cart.listCart.loading,
  attributeName: state.cart.attributeName.attributeName,
});

const mapDispatchToProps = (dispatch) => ({
  login: (data) => dispatch(login(data)),
  push: (url, item) => dispatch(push(url, item)),
  fetchCartList: (token) => dispatch(listCart(token)),
  fetchCartByUserId: (userId, token) => dispatch(searchByUserId(userId, token)),
  fetchCartListByCartId: (cartId, token) => dispatch(getListCartByCartId(cartId, token)),
  fetchPutCartListByCartId: (cartId, data, token) => dispatch(putListCartByCartId(cartId, data, token)),
  fetchPurchaseSummary: (cartId, token) => dispatch(getPurchaseSummary(cartId, token)),
  fetchStock: (token, body) => dispatch(getCheckStock(token, body)),
  deleteCart: (cartId, body, token) => dispatch(deleteAllCart(cartId, body, token)),
  deleteCartDetail: (cartDetailId, token) => dispatch(deleteCartDetail(cartDetailId, token)),
  fetchGetBranchList: (token) => dispatch(getBranchList(token)),
  fetchAttributeName: (name) => dispatch(getAttributeName(name)),
});

export default connect(mapStateToProps, mapDispatchToProps)(BucketListPage);
