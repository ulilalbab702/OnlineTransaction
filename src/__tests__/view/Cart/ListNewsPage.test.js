import React from "react";
import toJson from "enzyme-to-json";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import BucketListPage from "views/BucketListPage/BucketListPage";
import { cartDummyList } from "./../../../dummy/Cart.dummy";

Enzyme.configure({ adapter: new Adapter() });
describe("Cart List Component", () => {
    const wrapper = shallow(
        <BucketListPage
                cartList={cartDummyList}
        />
    );
    describe("Cart List Component", () => {
        it("Should match to snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});