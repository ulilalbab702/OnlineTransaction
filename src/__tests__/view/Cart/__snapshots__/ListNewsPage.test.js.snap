// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart List Component Cart List Component Should match to snapshot 1`] = `
<Fragment>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <ModalShippment
      isClose={[Function]}
      isOpen={false}
    />
    <ModalSelectBranch
      isClose={[Function]}
      isOpen={false}
      selectAddress={[Function]}
    />
    <Col
      className="containerBucket"
      lg="12"
      md="7"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xl="7"
    >
      <div>
        <h3
          className="styleBLBold"
        >
          Branch Supply Point
        </h3>
        <div
          className="cardBranch"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 10,
            }
          }
        >
          <div
            className="card-bullet"
          >
            <img
              height="15"
              src="iconWarehouse.png"
              width="15"
            />
          </div>
          <h5
            className="titleSelect"
          >
            Select Branch Supply Point
          </h5>
          <div
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <div
              className="btnSelectBranch"
              onClick={[Function]}
            >
              <p
                className="titleCheckout"
              >
                Select Branch
              </p>
            </div>
          </div>
        </div>
      </div>
      <Row
        className="headerBucketList"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <h3
            className="styleBLBold"
          >
            Bucket List
          </h3>
        </Col>
        <WithStyles(ForwardRef(Paper))
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3DA7DE",
              "borderRadius": "8px",
              "boxShadow": "none",
              "display": "flex",
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "width": 149,
            }
          }
        >
          <img
            className="iconDownloadList"
            src="iconDownload.png"
          />
          <p
            className="titleBtnDownloadList"
            style={
              Object {
                "color": "#ffffff",
                "marginBottom": 0,
                "marginLeft": 10,
                "textAlign": "center",
              }
            }
          >
            Download List
          </p>
        </WithStyles(ForwardRef(Paper))>
      </Row>
      <Row
        className="bucketList"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          className="productList"
          key="0"
          lg="12"
          md="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xl="12"
        >
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              style={
                Object {
                  "paddingLeft": "3px",
                }
              }
            >
              <WithStyles(ForwardRef(Checkbox))
                name="600-311-8291"
                onChange={[Function]}
                onClick={[Function]}
                style={
                  Object {
                    "color": "#5ACC4D",
                  }
                }
                value="checked"
              />
            </div>
            <div
              className="imageProduct"
            >
              <img
                className="imageList"
                src="https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa"
              />
            </div>
            <Col
              className="pl-0 productDescription"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <p
                className="mb-1 titleKategoriProductBL"
              >
                600-311-8291
              </p>
              <p
                className="titleProductBL"
              />
              <p
                className="mb-1 titlePriceBL"
              >
                IDR
                 
                500.000
              </p>
              <p
                className="mb-1 titlePriceDiscountBL"
              >
                IDR
                 
                300.000
              </p>
              <Row
                className="cardMinPlusBL"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="btnMin"
                >
                  <TiMinus />
                </div>
                <div
                  className="totalOrder"
                >
                  <h4
                    className="mb-0 teksTotalOrder"
                  >
                    2
                  </h4>
                </div>
                <div
                  className="btnPlus"
                >
                  <TiPlus />
                </div>
              </Row>
            </Col>
            <Col
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                }
              }
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="cardLabel"
              >
                <h5
                  style={
                    Object {
                      "color": "#148F0F",
                      "fontSize": 12,
                      "fontWeight": "bold",
                      "marginBottom": 0,
                    }
                  }
                >
                  Available
                </h5>
              </div>
            </Col>
          </Row>
        </Col>
        <Col
          className="productList"
          key="1"
          lg="12"
          md="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xl="12"
        >
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              style={
                Object {
                  "paddingLeft": "3px",
                }
              }
            >
              <WithStyles(ForwardRef(Checkbox))
                name="568002"
                onChange={[Function]}
                onClick={[Function]}
                style={
                  Object {
                    "color": "#5ACC4D",
                  }
                }
                value="checked"
              />
            </div>
            <div
              className="imageProduct"
            >
              <img
                className="imageList"
                src="https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa"
              />
            </div>
            <Col
              className="pl-0 productDescription"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <p
                className="mb-1 titleKategoriProductBL"
              >
                568002
              </p>
              <p
                className="titleProductBL"
              />
              <p
                className="mb-1 titlePriceBL"
              >
                IDR
                 
                1.000.000
              </p>
              <p
                className="mb-1 titlePriceDiscountBL"
              >
                IDR
                 
                800.000
              </p>
              <Row
                className="cardMinPlusBL"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="btnMin"
                >
                  <TiMinus />
                </div>
                <div
                  className="totalOrder"
                >
                  <h4
                    className="mb-0 teksTotalOrder"
                  >
                    2
                  </h4>
                </div>
                <div
                  className="btnPlus"
                >
                  <TiPlus />
                </div>
              </Row>
            </Col>
            <Col
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                }
              }
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="cardLabel"
              >
                <h5
                  style={
                    Object {
                      "color": "#148F0F",
                      "fontSize": 12,
                      "fontWeight": "bold",
                      "marginBottom": 0,
                    }
                  }
                >
                  Available
                </h5>
              </div>
            </Col>
          </Row>
        </Col>
        <hr
          style={
            Object {
              "margin": "0",
              "width": "100%",
            }
          }
        />
      </Row>
    </Col>
    <Col
      className="containerPromo"
      lg="12"
      md="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xl="5"
    >
      <div
        className="cardPromotion"
      >
        <Col
          className="p-0"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            className="kontenCardPromotion"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Col
              className="pl-0 teksCardPromotion"
              lg="12"
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Promotion
            </Col>
            <Col
              className="pl-0 pr-0 inputPromoOD"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <WithStyles(ForwardRef(Paper))
                className="inputPromoCode"
                elevation={1}
              >
                <WithStyles(ForwardRef(InputBase))
                  className="promoCode-text"
                  placeholder="Type Promo Code"
                />
                <Col
                  style={
                    Object {
                      "paddingLeft": "25px",
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="btnApply"
                  >
                    <p
                      className="teksApply"
                    >
                      Apply
                    </p>
                  </div>
                </Col>
              </WithStyles(ForwardRef(Paper))>
            </Col>
            <hr
              style={
                Object {
                  "marginBottom": "1rem",
                  "marginTop": "1.5rem",
                  "width": "100%",
                }
              }
            />
            <Col
              className="pl-0 teksCardOrderDetail"
              lg="12"
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Purchase Summary
            </Col>
            <Col
              className="p-0 "
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Row
                className="subTotal"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="teksSubTotal"
                >
                  Sub Total
                </div>
                <div
                  className="hargaSubTotal"
                >
                  2.500.000
                </div>
              </Row>
              <Row
                className="subTotal"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="teksSubTotal"
                >
                  Biaya Pengiriman
                </div>
                <div
                  className="hargaSubTotal"
                >
                  Not Yet Calculated
                </div>
              </Row>
              <hr
                style={
                  Object {
                    "borderTop": "1px dashed #707070",
                    "marginBottom": "0",
                    "marginTop": "1rem",
                    "opacity": "0.5",
                    "width": "100%",
                  }
                }
              />
              <Row
                className="subTotal"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="teksSubTotal"
                >
                  Total Pembayaran
                </div>
                <div
                  className="hargaSubTotal1"
                >
                  2.500.000
                </div>
              </Row>
            </Col>
          </Col>
          <Col
            className="bgCheckoutBL"
            lg="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="btnCheckoutBL"
              onClick={[Function]}
            >
              <p
                className="titleCheckout"
              >
                Check Out
              </p>
            </div>
          </Col>
        </Col>
      </div>
      <div>
        <div
          className="cardNotif"
        >
          <div>
            <img
              height="50"
              src="cautionYellow.png"
              width="50"
            />
          </div>
          <h6
            className="titleNotif"
          >
            There is item was not available, it will be impact for delay in delivery process, are you sure to continue transaction?
          </h6>
        </div>
      </div>
    </Col>
  </Row>
</Fragment>
`;
